addon-test-support/test-link.ts

```diff
@@ -0,0 +1,68 @@
+import { getOwner, setOwner } from '@ember/application';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+
+import { LinkParams } from 'ember-link/link';
+
+import LinkManagerService from 'dummy/services/link-manager';
+
+export default class TestLink {
+  private _params: LinkParams;
+
+  // Overwritable properties
+  @tracked isActive = false;
+  @tracked isActiveWithoutQueryParams = false;
+  @tracked isActiveWithoutModels = false;
+  @tracked isEntering = false;
+  @tracked isExiting = false;
+  @tracked url = guidFor(this);
+
+  // Event handlers
+  onTransitionTo?(): void;
+  onReplaceWith?(): void;
+
+  constructor(linkManager: LinkManagerService, params: LinkParams) {
+    setOwner(this, getOwner(linkManager));
+    this._params = params;
+  }
+
+  get routeName() {
+    return this._params.route;
+  }
+
+  get qualifiedRouteName() {
+    return this.routeName;
+  }
+
+  get models() {
+    return this._params.models;
+  }
+
+  get queryParams() {
+    return this._params.query;
+  }
+
+  @action
+  transitionTo(event: Event) {
+    this._preventTransitionOut(event);
+
+    if (this.onTransitionTo) {
+      this.onTransitionTo();
+    }
+  }
+
+  @action
+  replaceWith(event: Event) {
+    this._preventTransitionOut(event);
+
+    if (this.onReplaceWith) {
+      this.onReplaceWith();
+    }
+  }
+
+  private _preventTransitionOut(event: Event) {
+    // Make sure we don't transition out of the testing page
+    event && event.preventDefault();
+  }
```

Comment:
Should we rather apply the same safe-guards here, so we don't cancel _any_ event?

Reply:
Ignore me, I thought this does more like checking for the `target.tagName` etc.

https://github.com/buschtoens/ember-link/blob/b8dce02499506f1d02f4aa76b1fb91107389d6a7/addon/link.ts#L260-L267